synapseclient/models/agent.py

```diff
@@ -0,0 +1,629 @@
+    response: Optional[str] = None
+    """The response from the agent."""
+
+    enable_trace: Optional[bool] = False
+    """Whether tracing is enabled for the prompt."""
+
+    trace: Optional[str] = None
+    """The trace or "thought process" of the agent when responding to the prompt."""
+
+    def to_synapse_request(self):
+        """Converts the request to a request expected of the Synapse REST API."""
+        return {
+            "concreteType": self.concrete_type,
+            "sessionId": self.session_id,
+            "chatText": self.prompt,
+            "enableTrace": self.enable_trace,
+        }
+
+    def fill_from_dict(self, synapse_response: Dict[str, str]) -> "AgentPrompt":
+        """
+        Converts a response from the REST API into this dataclass.
+
+        Arguments:
+            synapse_response: The response from the REST API.
+
+        Returns:
+            The AgentPrompt object.
+        """
+        self.id = synapse_response.get("jobId", None)
+        self.session_id = synapse_response.get("sessionId", None)
+        self.response = synapse_response.get("responseText", None)
+        return self
+
+    async def _post_exchange_async(
+        self, *, synapse_client: Optional[Synapse] = None, **kwargs
+    ) -> None:
+        """Retrieves information about the trace of this prompt with the agent.
+
+        Arguments:
+            synapse_client: If not passed in and caching was not disabled by
+                    `Synapse.allow_client_caching(False)` this will use the last created
+                    instance from the Synapse class constructor.
+        """
+        if self.enable_trace:
+            trace_response = await get_trace(
+                prompt_id=self.id,
+                newer_than=kwargs.get("newer_than", None),
+                synapse_client=synapse_client,
+            )
+            self.trace = trace_response["page"][0]["message"]
+
+
+@dataclass
+@async_to_sync
+class AgentSession(AgentSessionSynchronousProtocol):
+    """Represents a [Synapse Agent Session](https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/agent/AgentSession.html)
+
+    Attributes:
+        id: The unique ID of the agent session.
+            Can only be used by the user that created it.
+        access_level: The access level of the agent session.
+            One of PUBLICLY_ACCESSIBLE, READ_YOUR_PRIVATE_DATA,
+            or WRITE_YOUR_PRIVATE_DATA.
+        started_on: The date the agent session was started.
+        started_by: The ID of the user who started the agent session.
+        modified_on: The date the agent session was last modified.
+        agent_registration_id: The registration ID of the agent that will
+            be used for this session.
+        etag: The etag of the agent session.
+
+    Note: It is recommended to use the `Agent` class to start and get sessions,
+    but you are free to use this class directly if you wish.
+
+    Example: Start a session and send a prompt.
+
```

Comment:
The formatting of these examples isn't exactly correct:
https://synapsepythonclient--1152.org.readthedocs.build/en/1152/reference/experimental/sync/agent/#synapseclient.models.Agent

![image](https://github.com/user-attachments/assets/0ad2cc80-5f79-46c7-842a-2b5de4862507)

It must be in this format (Copied from another spot in the code):

```
        Example: Using this function
            Download file into cache

                from synapseclient import Synapse
                from synapseclient.api import get_from_entity_factory

                syn = Synapse()
                syn.login()

                entity = await get_from_entity_factory(synapse_id_or_path='syn1906479')
                print(entity.name)
                print(entity.path)
```